tests/tests_dyndns.yml

```
@@ -0,0 +1,144 @@
+# and the DNS records from any previous runs removed.
+
+# Example inventory:
+# [client]
+# client1 ansible_connection=local ansible_host=127.0.0.1 \
+#  ad_integration_join_to_dc="ad1.domain.com" \
+#  ad_integration_password=Secret123 ad_integration_realm="domain.com" \
+#  ad_dyndns_update=true
+# [ad]
+# ad1 ansible_host=<AD IP> ansible_connection=winrm ansible_password=Secret123\
+#  ansible_port=5986 ansible_user=Administrator \
+#  ansible_winrm_server_cert_validation=ignore
+
+- name: Ensure that the role configures dynamic dns
+  hosts: client
+  gather_facts: false  # test that role works in this case
+
+  tasks:
+    - name: Test - Run the system role with dumby vars
+      include_role:
+        name: linux-system-roles.ad_integration
+      vars:
+        ad_dyndns_iface: "TESTING"
+        ad_dyndns_server: 127.0.0.1
+
+    - name: Test - Verify sssd.conf options were written
+      block:
+        - name: Copy sssd.conf file from host to local /tmp/
+          fetch:
+            src: /etc/sssd/sssd.conf
+            dest: /tmp/
+            flat: true
+          changed_when: false
+        - name: Assert sssd.conf options were written
+          assert:
+            that:
+              - "{{ lookup('ini', 'dyndns_update', section='domain/' +
+                ad_integration_realm | lower, file='/tmp/sssd.conf') }} == True"
+              - "{{ lookup('ini', 'dyndns_ttl', section='domain/' +
+                ad_integration_realm | lower, file='/tmp/sssd.conf') }} == 3600"
+              - "'{{ lookup('ini', 'dyndns_iface', section='domain/' +
+                ad_integration_realm | lower, file='/tmp/sssd.conf') }}'
+                == 'TESTING'"
+              - "{{ lookup('ini', 'dyndns_refresh_interval', section='domain/' +
+                ad_integration_realm | lower, file='/tmp/sssd.conf') }}
+                == 86400"
+              - "{{ lookup('ini', 'dyndns_update_ptr', section='domain/' +
+                ad_integration_realm | lower, file='/tmp/sssd.conf') }} == True"
+              - "{{ lookup('ini', 'dyndns_force_tcp', section='domain/' +
+                ad_integration_realm | lower, file='/tmp/sssd.conf') }}
+                == false"
+              - "'{{ lookup('ini', 'dyndns_auth', section='domain/' +
+                ad_integration_realm | lower, file='/tmp/sssd.conf') }}'
+                == 'GSS-TSIG'"
+              - "'{{ lookup('ini', 'dyndns_server', section='domain/' +
+                ad_integration_realm | lower, file='/tmp/sssd.conf') }}'
+                == '127.0.0.1'"
+        - name: Remove /tmp/sssd.conf
+          file:
+            path: /tmp/sssd.conf
+            state: absent
+          changed_when: false
+
+    - name: Test - Check sssd.log for unsupported options
+      block:
+        - name: Search /var/log/sssd/sssd.log for [sss_ini_call_validators]
+          # noqa: command-instead-of-shell
+          shell: "grep -i 'sss_ini_call_validators' /var/log/sssd/sssd.log"
```

```suggestion
          command: grep -i sss_ini_call_validators /var/log/sssd/sssd.log
```
better to use `command` instead of `shell` if you aren't using redirection, piping, etc.